packages/salesforcedx-vscode-core/src/commands/templates/libraryBaseTemplateCommand.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  TemplateOptions,
+  TemplateService,
+  TemplateType
+} from '@salesforce/templates';
+import {
+  CommandletExecutor,
+  PathStrategyFactory,
+  SelectOutputDir,
+  SourcePathStrategy
+} from '../util';
+
+import { channelService } from '../../channels';
+import { notificationService } from '../../notifications';
+import { telemetryService } from '../../telemetry';
+import {
+  getRootWorkspacePath,
+  hasRootWorkspace,
+  MetadataDictionary,
+  MetadataInfo
+} from '../../util';
+
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+
+import * as path from 'path';
+import { ProgressLocation, window, workspace } from 'vscode';
+
+interface ExecutionResult {
+  output?: string;
+  error?: Error;
+}
+
+/**
+ * Warpping function exeuction with VS Code interface, such as Output channel and progress
+ * @param commandName command name
+ * @param fn function to execute
+ */
+function wrapExecute(
```

Comment:
We could move this to be part of `LibraryBaseTemplateCommand` and would make it easier reference some of it's data and functionality

Reply:
The reason I have it as a separate function is I hope it could be reusable for other commands and not limited to `LibraryBaseTemplateCommand`

Reply:
Sure, but it's a bit early to know if that will happen since there's still more work to be done in the other libraries. It's better if we generalize the code once we have the other libraries ready. For now I want to make sure it's easy to maintain and finish the rest of the work for templates

Reply:
Well I've extracted the function out precisely because I couldn't reuse similar `wrapExecute` function from other libraries. I'm not sure if it's still worth wrapping with similar functions if it's only used once. (vs. just inlining what's in the function).

The rest of the work for templates doesn't require more changes from `LibraryBaseTemplateCommand`, since they are all extending the base class, except for error telemetry.

Reply:
For example `getLogsWrapper`, `retrieveWrapper` are only used once in apex/sdr.

Reply:
Got it, those libraries are still going to change which is why it might not be a good moment to generalize. 

If there's not much to re-use I think we can get rid of the wrapper function here and just keep everything inside of `execute`. It would also make the code easier to read.

Reply:
Sounds good, if you don't think it can be used elsewhere for now, I can move what's inside the wrapper function into `execute`.

Reply:
I don't see that happening in the next weeks so it's safe to make the update.

Reply:
Cool commit https://github.com/forcedotcom/salesforcedx-vscode/pull/2428/commits/5a6e09d49ff584df8aa098da06288bb9e3db4dc5 removes `wrapExecute`.